Eslint not work. Downloading eslint failed

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Eslint not work.

Display below info in loop about 3 times when  open .ts file.

![image](https://github.com/zed-industries/zed/assets/17584899/22722b65-0123-4040-b629-4a0e7a0bd112)


### Environment

Zed: v0.135.2 (Zed)
OS: macOS 11.5.2
Memory: 16 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![image](https://github.com/zed-industries/zed/assets/17584899/98a4db2f-798d-45ae-9a82-4586d8d84d10)


### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

2024-05-16T11:29:28+08:00 [INFO] ========== starting zed ==========
2024-05-16T11:29:28+08:00 [INFO] Opening main db
2024-05-16T11:29:28+08:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2024-05-16T11:29:28+08:00 [INFO] extensions updated. loading 7, reloading 0, unloading 0
2024-05-16T11:29:28+08:00 [INFO] Opening main db
2024-05-16T11:29:28+08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/yol/Documents/android-sdk-macosx/tools:/Users/yol/Documents/android-sdk-macosx/platform-tools:/Users/yol/.npm-global/bin:/Users/yol/.config/yarn/global/node_modules/.bin:/usr/local/mongodb/bin:/usr/local/bin:/usr/local/sbin:/Library/Frameworks/Python.framework/Versions/3.8/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/yol/.cargo/bin
2024-05-16T11:29:29+08:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-05-16T11:29:29+08:00 [INFO] Opening main db
2024-05-16T11:29:29+08:00 [INFO] Opening main db
2024-05-16T11:29:29+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:29+08:00 [INFO] starting language servers for JSON: json-language-server
2024-05-16T11:29:29+08:00 [INFO] starting language server "json-language-server", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 1
2024-05-16T11:29:29+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:29+08:00 [INFO] starting language servers for JSON: json-language-server
2024-05-16T11:29:29+08:00 [INFO] fetching latest version of language server "json-language-server"
2024-05-16T11:29:29+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:29+08:00 [ERROR] crates/workspace/src/persistence/model.rs:313: No worktree for path: "/Users/yol/.config/zed/settings.json"
2024-05-16T11:29:30+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:30+08:00 [INFO] starting language servers for Vue.js: tailwindcss-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] starting language server "tailwindcss-language-server", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 2
2024-05-16T11:29:30+08:00 [INFO] starting language server "eslint", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 3
2024-05-16T11:29:30+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:30+08:00 [INFO] starting language servers for TypeScript: typescript-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] starting language server "typescript-language-server", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 4
2024-05-16T11:29:30+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:30+08:00 [INFO] starting language servers for JSON: json-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language servers for Vue.js: tailwindcss-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] starting language servers for TypeScript: typescript-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:30+08:00 [INFO] starting language servers for JSON: json-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language servers for Vue.js: tailwindcss-language-server, eslint, vue-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language server "vue-language-server", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 5
2024-05-16T11:29:30+08:00 [INFO] starting language servers for TypeScript: typescript-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] fetching latest version of language server "tailwindcss-language-server"
2024-05-16T11:29:30+08:00 [INFO] fetching latest version of language server "eslint"
2024-05-16T11:29:30+08:00 [INFO] downloading language server "eslint"
2024-05-16T11:29:30+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:30+08:00 [INFO] starting language servers for JSON: json-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language servers for Vue.js: tailwindcss-language-server, eslint, vue-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language servers for TypeScript: typescript-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] fetching latest version of language server "typescript-language-server"
2024-05-16T11:29:30+08:00 [INFO] Initializing default prettier with plugins {}
2024-05-16T11:29:30+08:00 [INFO] starting language servers for JSON: json-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language servers for Vue.js: tailwindcss-language-server, eslint, vue-language-server
2024-05-16T11:29:30+08:00 [INFO] starting language servers for TypeScript: typescript-language-server, eslint
2024-05-16T11:29:30+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:30+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:31+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:31+08:00 [INFO] downloading language server "json-language-server"
2024-05-16T11:29:32+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:32+08:00 [INFO] downloading language server "tailwindcss-language-server"
2024-05-16T11:29:32+08:00 [INFO] starting language server. binary path: "/Users/yol/Library/Application Support/Zed/node/node-v18.15.0-darwin-x64/bin/node", working directory: "/Users/yol/Documents/project/uni-ketogenesis", args: ["/Users/yol/Library/Application Support/Zed/languages/json-language-server/node_modules/vscode-json-languageserver/bin/vscode-json-languageserver", "--stdio"]
2024-05-16T11:29:32+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:33+08:00 [INFO] starting language server. binary path: "/Users/yol/Library/Application Support/Zed/node/node-v18.15.0-darwin-x64/bin/node", working directory: "/Users/yol/Documents/project/uni-ketogenesis", args: ["/Users/yol/Library/Application Support/Zed/languages/tailwindcss-language-server/node_modules/.bin/tailwindcss-language-server", "--stdio"]
2024-05-16T11:29:33+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:34+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:35+08:00 [INFO] starting language server. binary path: "/Users/yol/Library/Application Support/Zed/node/node-v18.15.0-darwin-x64/bin/node", working directory: "/Users/yol/Documents/project/uni-ketogenesis", args: ["/Users/yol/Library/Application Support/Zed/extensions/work/vue/node_modules/@vue/language-server/bin/vue-language-server.js", "--stdio"]
2024-05-16T11:29:35+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-05-16T11:29:35+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:35+08:00 [ERROR] failed to start language server "eslint": error downloading release: request or operation took longer than the configured timeout time
2024-05-16T11:29:35+08:00 [ERROR] server stderr: Some("")
2024-05-16T11:29:35+08:00 [INFO] retrying installation of language server "eslint" in 1s
2024-05-16T11:29:35+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-05-16T11:29:35+08:00 [ERROR] crates/copilot/src/copilot.rs:1009: error downloading copilot release

Caused by:
    0: request or operation took longer than the configured timeout time
    1: [28] Timeout was reached
2024-05-16T11:29:35+08:00 [INFO] downloading language server "typescript-language-server"
2024-05-16T11:29:35+08:00 [INFO] Node runtime install_if_needed
2024-05-16T11:29:36+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:36+08:00 [INFO] About to spawn test binary
2024-05-16T11:29:36+08:00 [WARN] test binary failed to launch
2024-05-16T11:29:36+08:00 [WARN] test binary check failed
2024-05-16T11:29:36+08:00 [INFO] beginning to reinstall server
2024-05-16T11:29:36+08:00 [INFO] deleting server container
2024-05-16T11:29:36+08:00 [INFO] starting language server "eslint", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 6
2024-05-16T11:29:36+08:00 [INFO] fetching latest version of language server "eslint"
2024-05-16T11:29:36+08:00 [INFO] downloading language server "eslint"
2024-05-16T11:29:36+08:00 [INFO] starting language server. binary path: "/Users/yol/Library/Application Support/Zed/node/node-v18.15.0-darwin-x64/bin/node", working directory: "/Users/yol/Documents/project/uni-ketogenesis", args: ["/Users/yol/Library/Application Support/Zed/languages/typescript-language-server/node_modules/typescript-language-server/lib/cli.mjs", "--stdio"]
2024-05-16T11:29:36+08:00 [INFO] Language server with id 4 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "Using Typescript version (user-setting) 4.9.5 from path \"/Users/yol/Documents/project/uni-ketogenesis/node_modules/typescript/lib/tsserver.js\""
}
2024-05-16T11:29:36+08:00 [INFO] Language server with id 4 sent unhandled notification $/typescriptVersion:
{
  "version": "4.9.5",
  "source": "user-setting"
}
2024-05-16T11:29:41+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-05-16T11:29:41+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:41+08:00 [ERROR] failed to start language server "eslint": error downloading release: request or operation took longer than the configured timeout time
2024-05-16T11:29:41+08:00 [ERROR] server stderr: Some("")
2024-05-16T11:29:41+08:00 [INFO] retrying installation of language server "eslint" in 1s
2024-05-16T11:29:42+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:42+08:00 [INFO] About to spawn test binary
2024-05-16T11:29:42+08:00 [WARN] test binary failed to launch
2024-05-16T11:29:42+08:00 [WARN] test binary check failed
2024-05-16T11:29:42+08:00 [INFO] beginning to reinstall server
2024-05-16T11:29:42+08:00 [INFO] deleting server container
2024-05-16T11:29:42+08:00 [INFO] starting language server "eslint", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 7
2024-05-16T11:29:42+08:00 [INFO] fetching latest version of language server "eslint"
2024-05-16T11:29:42+08:00 [INFO] downloading language server "eslint"
2024-05-16T11:29:47+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-05-16T11:29:47+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:47+08:00 [ERROR] failed to start language server "eslint": error downloading release: request or operation took longer than the configured timeout time
2024-05-16T11:29:47+08:00 [ERROR] server stderr: Some("")
2024-05-16T11:29:47+08:00 [INFO] retrying installation of language server "eslint" in 1s
2024-05-16T11:29:48+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:48+08:00 [INFO] About to spawn test binary
2024-05-16T11:29:48+08:00 [WARN] test binary failed to launch
2024-05-16T11:29:48+08:00 [WARN] test binary check failed
2024-05-16T11:29:48+08:00 [INFO] beginning to reinstall server
2024-05-16T11:29:48+08:00 [INFO] deleting server container
2024-05-16T11:29:48+08:00 [INFO] starting language server "eslint", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 8
2024-05-16T11:29:48+08:00 [INFO] fetching latest version of language server "eslint"
2024-05-16T11:29:48+08:00 [INFO] downloading language server "eslint"
2024-05-16T11:29:53+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-05-16T11:29:53+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:53+08:00 [ERROR] failed to start language server "eslint": error downloading release: request or operation took longer than the configured timeout time
2024-05-16T11:29:53+08:00 [ERROR] server stderr: Some("")
2024-05-16T11:29:53+08:00 [INFO] retrying installation of language server "eslint" in 1s
2024-05-16T11:29:54+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:54+08:00 [INFO] About to spawn test binary
2024-05-16T11:29:54+08:00 [WARN] test binary failed to launch
2024-05-16T11:29:54+08:00 [WARN] test binary check failed
2024-05-16T11:29:54+08:00 [INFO] beginning to reinstall server
2024-05-16T11:29:54+08:00 [INFO] deleting server container
2024-05-16T11:29:54+08:00 [INFO] starting language server "eslint", path: "/Users/yol/Documents/project/uni-ketogenesis", id: 9
2024-05-16T11:29:54+08:00 [INFO] fetching latest version of language server "eslint"
2024-05-16T11:29:54+08:00 [INFO] downloading language server "eslint"
2024-05-16T11:29:59+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-05-16T11:29:59+08:00 [ERROR] crates/languages/src/typescript.rs:427: missing first file
2024-05-16T11:29:59+08:00 [ERROR] failed to start language server "eslint": error downloading release: request or operation took longer than the configured timeout time
2024-05-16T11:29:59+08:00 [ERROR] server stderr: Some("")
2024-05-16T11:29:59+08:00 [ERROR] Hit 4 reinstallation attempts for "eslint"
2024-05-16T11:31:22+08:00 [INFO] reload git repository ".git"
2024-05-16T11:31:22+08:00 [INFO] Prettier config file "package.json" changed, reloading prettier instances for worktree 4294967311
2024-05-16T11:31:46+08:00 [INFO] reload git repository ".git"
2024-05-16T11:31:46+08:00 [INFO] reload git repository ".git"
2024-05-16T11:33:21+08:00 [ERROR] crates/lsp/src/lsp.rs:773: oneshot canceled
2024-05-16T11:33:21+08:00 [ERROR] crates/lsp/src/lsp.rs:773: oneshot canceled
2024-05-16T11:33:21+08:00 [ERROR] crates/lsp/src/lsp.rs:773: oneshot canceled
2024-05-16T11:33:21+08:00 [ERROR] crates/lsp/src/lsp.rs:773: oneshot canceled

It seems that I made a mistake. I actually downloaded and installed `TypeScriptLspAdapter` not the `EsLintLspAdapter`.

I retested, the pkg can be downloaded via a proxy, but can not install.

The `EsLintLspAdapter` fails at installing:

https://github.com/zed-industries/zed/blob/410c46a5517e397890580c21dab13575e7839d5a/crates/languages/src/typescript.rs#L382

The error message:
```rust
server stderr: Some("
    node:internal/modules/cjs/loader:1078
    throw err;
    Error: Cannot find module 'C:\\Users\\xiaobai\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
Node.js v18.15.0
")
```

The problems are:
1. The actuall install path is `...\Zed\languages\eslint\vscode-eslint-release\2.4.4\vscode-eslint`, but it requires `...\Zed\languages\eslint\vscode-eslint-release\vscode-eslint`
2. The file `\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js` does not exist.

I can reproduce this error both on windows and macOS.

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -226,18 +226,18 @@ impl NodeRuntime for RealNodeRuntime {
 
             let node_binary = installation_path.join(NODE_PATH);
             let npm_file = installation_path.join(NPM_PATH);
-            let mut env_path = vec![node_binary
+            let mut env_path = vec![vec![node_binary
                 .parent()
                 .expect("invalid node binary path")
-                .to_path_buf()];
+                .to_path_buf()]];
 
             if let Some(existing_path) = std::env::var_os("PATH") {
-                if !existing_path.is_empty() {
-                    let paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
-                    env_path.extend(paths);
-                }
+                let mut paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
+                env_path.append(&mut paths);
             }
-            let env_path = std::env::join_paths(&env_path)?;
+
+            let env_path =
+                std::env::join_paths(env_path).context("failed to create PATH env variable")?;
 
             if smol::fs::metadata(&node_binary).await.is_err() {
                 return Err(anyhow!("missing node binary file"));

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -227,17 +227,18 @@ impl NodeRuntime for RealNodeRuntime {
             let node_binary = installation_path.join(NODE_PATH);
             // std::env::set_var(key, value);
             let npm_file = installation_path.join(NPM_PATH);
-            let mut env_path = node_binary
+            let env_path = node_binary
                 .parent()
                 .expect("invalid node binary path")
                 .to_path_buf();
+            let mut env_path = env_path.to_string_lossy().to_string();
             println!("============================================");
-            println!("  Env path: {}", env_path.display());
+            println!("  Env path: {}", env_path);
 
             if let Some(existing_path) = std::env::var_os("PATH") {
                 if !existing_path.is_empty() {
-                    env_path.push(":");
-                    env_path.push(&existing_path);
+                    env_path.push_str(":");
+                    env_path.push_str(existing_path.to_str().unwrap());
                 }
             }
 
@@ -249,7 +250,7 @@ impl NodeRuntime for RealNodeRuntime {
                 return Err(anyhow!("missing npm file"));
             }
             println!("============================================");
-            println!("  Env path: {}", env_path.display());
+            println!("  Env path: {}", env_path);
 
             let mut command = Command::new(node_binary);
             command.env_clear();

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -232,8 +232,6 @@ impl NodeRuntime for RealNodeRuntime {
                 .expect("invalid node binary path")
                 .to_path_buf();
             let mut env_path = env_path.to_string_lossy().to_string();
-            println!("============================================");
-            println!("  Env path: {}", env_path);
 
             if let Some(existing_path) = std::env::var_os("PATH") {
                 if !existing_path.is_empty() {
@@ -249,8 +247,6 @@ impl NodeRuntime for RealNodeRuntime {
             if smol::fs::metadata(&npm_file).await.is_err() {
                 return Err(anyhow!("missing npm file"));
             }
-            println!("============================================");
-            println!("  Env path: {}", env_path);
 
             let mut command = Command::new(node_binary);
             command.env_clear();

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -225,7 +225,6 @@ impl NodeRuntime for RealNodeRuntime {
             let installation_path = self.install_if_needed().await?;
 
             let node_binary = installation_path.join(NODE_PATH);
-            // std::env::set_var(key, value);
             let npm_file = installation_path.join(NPM_PATH);
             let env_path = node_binary
                 .parent()

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -226,18 +226,18 @@ impl NodeRuntime for RealNodeRuntime {
 
             let node_binary = installation_path.join(NODE_PATH);
             let npm_file = installation_path.join(NPM_PATH);
-            let env_path = node_binary
+            let mut env_path = vec![node_binary
                 .parent()
                 .expect("invalid node binary path")
-                .to_path_buf();
-            let mut env_path = env_path.to_string_lossy().to_string();
+                .to_path_buf()];
 
             if let Some(existing_path) = std::env::var_os("PATH") {
                 if !existing_path.is_empty() {
-                    env_path.push_str(":");
-                    env_path.push_str(existing_path.to_str().unwrap());
+                    let paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
+                    env_path.extend(paths);
                 }
             }
+            let env_path = std::env::join_paths(&env_path)?;
 
             if smol::fs::metadata(&node_binary).await.is_err() {
                 return Err(anyhow!("missing node binary file"));

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -225,11 +225,14 @@ impl NodeRuntime for RealNodeRuntime {
             let installation_path = self.install_if_needed().await?;
 
             let node_binary = installation_path.join(NODE_PATH);
+            // std::env::set_var(key, value);
             let npm_file = installation_path.join(NPM_PATH);
             let mut env_path = node_binary
                 .parent()
                 .expect("invalid node binary path")
                 .to_path_buf();
+            println!("============================================");
+            println!("  Env path: {}", env_path.display());
 
             if let Some(existing_path) = std::env::var_os("PATH") {
                 if !existing_path.is_empty() {
@@ -245,6 +248,8 @@ impl NodeRuntime for RealNodeRuntime {
             if smol::fs::metadata(&npm_file).await.is_err() {
                 return Err(anyhow!("missing npm file"));
             }
+            println!("============================================");
+            println!("  Env path: {}", env_path.display());
 
             let mut command = Command::new(node_binary);
             command.env_clear();

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -226,10 +226,10 @@ impl NodeRuntime for RealNodeRuntime {
 
             let node_binary = installation_path.join(NODE_PATH);
             let npm_file = installation_path.join(NPM_PATH);
-            let mut env_path = vec![vec![node_binary
+            let mut env_path = vec![node_binary
                 .parent()
                 .expect("invalid node binary path")
-                .to_path_buf()]];
+                .to_path_buf()];
 
             if let Some(existing_path) = std::env::var_os("PATH") {
                 let mut paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();

